cmd/crane/cmd/digest.go

```diff
@@ -24,16 +24,42 @@ import (
 
 // NewCmdDigest creates a new cobra.Command for the digest subcommand.
 func NewCmdDigest(options *[]crane.Option) *cobra.Command {
-	return &cobra.Command{
+	var tarball string
+	cmd := &cobra.Command{
 		Use:   "digest IMAGE",
 		Short: "Get the digest of an image",
-		Args:  cobra.ExactArgs(1),
-		Run: func(_ *cobra.Command, args []string) {
-			digest, err := crane.Digest(args[0], *options...)
-			if err != nil {
-				log.Fatalf("computing digest: %v", err)
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			if tarball != "" {
+				tag := ""
+				if len(args) > 0 {
+					tag = args[0]
+				}
+
+				img, err := crane.LoadTag(tarball, tag, *options...)
+				if err != nil {
+					log.Fatalf("loading image from %q: %v", tarball, err)
+				}
+				digest, err := img.Digest()
+				if err != nil {
+					log.Fatalf("computing digest: %v", err)
+				}
+				fmt.Println(digest.String())
```

Comment:
nit: return here, and unindent the `else` block. Maybe make both branches helper methods?

Reply:
You drive a hard bargain.